[Mohamed Noriman bin Mohamed Farid] iP

sw-boi opened the pull request:
## Introducing AlfredPro!
>“Your mind is for having ideas, not holding them.” – David Allen

AlfredPro frees your mind of having to remember things you need to do. It's,
 * text-based
 * easy to learn
 * ~FAST~ _SUPER_ FAST to use

All you need to do is,
 1. download it from [here](https://github.com/seaweediman/ip/releases/tag/v1.0)
 2. double-click it.
 3. add your tasks.
 4. let it manage tasks for you

And it is **FREE** :smiley: !

Features:

- [x] Manages to-dos, deadlines and events

- [x] Search function to find your tasks

- [ ] Pretty GUI (Coming Soon!)

Enter `deadline return book /by June 6th` and AlfredPro saves it!

```json
{
  "taskType": "deadline",
  "isDone": "false",
  "desc": "return book",
  "by:": "2021-07-26 8pm"
}
```

Commit:
Add Gradle support

sw-boi committed:
Level-1

sw-boi committed:
Level 2

sw-boi committed:
Fixed the index for list command

sw-boi committed:
level 3

sw-boi committed:
Added task class

sw-boi committed:
Finished Todo class

sw-boi committed:
Finished Level-4

sw-boi committed:
Finished Level-5

sw-boi committed:
Added delete

sw-boi committed:
Fixed the automated testing

sw-boi committed:
changed if-else statements to switch

sw-boi committed:
Used StringBuilder

sw-boi committed:
Change indent for switch-case

sw-boi committed:
Level-7 DONE

sw-boi committed:
before merge

sw-boi committed:
Level-7

sw-boi committed:
Level-8

sw-boi committed:
Finish part 1 of OOP

sw-boi committed:
Finish OOP

sw-boi committed:
Add packages

sw-boi committed:
Add JUNIT

sw-boi committed:
Add javadocs

sw-boi committed:
Added add function to storagE

sw-boi committed:
Add Storage methods

sw-boi committed:
Finish javadocs

sw-boi committed:
Fix javadoc

sw-boi committed:
Fix coding standards

sw-boi committed:
Add FindCommand

sw-boi committed:
Fix some javadocs

sw-boi committed:
make jar file

sw-boi committed:
Add support for dates for deadline and event

sw-boi committed:
Add gradle

sw-boi committed:
Add Gradle

sw-boi committed:
Merge branch 'add-gradle-support'

sw-boi committed:
Add Gradle

sw-boi committed:
Add CheckStyle

sw-boi committed:
Clean up code

sw-boi committed:
Clean up code

sw-boi committed:
Add date support

sw-boi committed:
Add gui

sw-boi committed:
Add GUI

sw-boi committed:
Touch up GUI

rish-16 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+
+/**
+ * Represents the chatbot Duke.
+ */
```
Public classes must have `@author` and `@version` in the JavaDoc comments under the main description

rish-16 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+            }
+            return new AddTodoCommand(desc);
+        case "deadline":
+            String by;
```
For short variable name declarations, consider adding a small inline comment describing it. Like so:

```java
String by; // deadline for Deadline Task
```

rish-16 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+ */
+public class TaskList {
+
+    private ArrayList<Task> list;
```
Consider naming `list` as something else that's more descriptive? like `tasks` or something along those lines.

rish-16 reviewed (commented):
LGTM 👍🏻  Few minor changes here and there. Widely follows the module's coding guide. The code is very modular and easy to follow.

lhw-1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    /**
+     * Method used to run the program.
```
Consider adding `@param` and `@return` tags to JavaDoc comments for public methods?

lhw-1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    /**
+     * Parses the date
```
JavaDoc comments can be more descriptive, e.g. "Parses the date string given by the user into a LocalDateTime object."

lhw-1 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Returns a list of task whose description includes the keyword.
+     * @param keyword
+     * @return
```
The JavaDoc `@return` tag should specify what is being returned.

lhw-1 reviewed (commented):
LGTM. Coding quality is up to standard, with a few minor changes already pointed out. Mostly follows the coding standard, with a few adjustments here and there. Good work!

janjanchen commented on src/main/java/Launcher.java:
```diff
@@ -0,0 +1,7 @@
+import javafx.application.Application;
+
+public class Launcher {
```
Header comments should be added for public classes and methods

janjanchen commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Marks a task as done and reflects it in the storage.
+     * @param index of task to be marked as done.
+     * @throws IOException
```
The Javadoc tag `@throws` should specify the condition for the exception to be thrown

janjanchen commented on src/main/java/duke/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,69 @@
+    /**
+     * Executes the command. Adds deadline to task list. Updates the save file.
+     *
+     * @param tasks the task list.
```
Javadoc tag `@throw` should be added after `@param`

janjanchen reviewed (commented):
LGTM :) Few minor changes could be made but the code is overall easy to read. Coding standard is also followed for most of the parts.

sw-boi committed:
Change color scheme of Duke

sw-boi committed:
Just a test

sw-boi committed:
Before making branch

sw-boi committed:
Add assertions to Duke.java

Assertions are used to define assumptions about the program state so
that the runtime can verify them. An assertion failure indicates a
possible bug in the code because the code has resulted in a program
state that violates an assumption about how the code should behave.

If at runtime, an assertion failure is detected, the program terminates
with and error message as a potential bug is detected. And sooner
it is detected, the safer it is.

sw-boi committed:
Improve code quality

Better code quality implies better readiblity.
Code that is more readible allows others who see the code
to understand how the code works clearly.

sw-boi committed:
Merge pull request #1 from seaweediman/branch-A-Assertions

Add assertions to Duke.java

sw-boi committed:
Merge pull request #2 from seaweediman/branch-A-CodeQuality

Improve code quality

sw-boi committed:
Remove isDone method in all commands

sw-boi committed:
Before swap

sw-boi committed:
Add way to perform tasks on multiple items

delete and done command only allow 1 index number.
That means if a user wants to delete or mark multiple tasks,
he or she will need to enter the command multiple times.

Let's give the ability to delete or mark multiple tasks in one go.

sw-boi committed:
Add ability to perform tasks on multiple items

sw-boi committed:
Add gradle.yml

Let's add the following features to make this a minimal project
template that students can use to start their project.
* Gradle support, a basic build script
* a skeleton JUnit test
* a basic CheckStyle configuration
* a basic I/O test setup
* support for GitHub actions
* documentation skeletons inside the docs folder

sw-boi committed:
Fix naming issues

sw-boi committed:
Narrow down date-time format

sw-boi committed:
Make window wider

sw-boi committed:
Add SLAP to Parser.java

sw-boi committed:
Add UserGuide

sw-boi committed:
Add Hyperlinks

sw-boi committed:
Add more hyperlinks for user-guide

sw-boi committed:
Clean up code

sw-boi committed:
Set theme jekyll-theme-midnight

sw-boi committed:
Set theme jekyll-theme-cayman

sw-boi committed:
Change README.md

sw-boi committed:
Merge branch 'master' of https://github.com/seaweediman/ip

sw-boi committed:
Update UG

sw-boi committed:
Update JUnit tests

sw-boi committed:
Clean up code

sw-boi committed:
Clean up code

sw-boi committed:
Update readme

sw-boi committed:
Update readme

sw-boi committed:
Update readme

sw-boi committed:
Update ReadMe.md

sw-boi committed:
update ReadME

sw-boi committed:
update readme

sw-boi committed:
Fix runtest.sh

sw-boi committed:
Fix runtest bug

sw-boi committed:
Update gradle.yml
